widgets/WaitingRoom.js

```diff
@@ -652,337 +652,4 @@
-     *    is a function, this parameter is passed to it, when invoked to get
-     *    a return value. Default: undefined
-     *
-     * @return {string} res The requested value.
-     *
-     * @see strGetter
-     */
-    function strGetterMulti(that, collection, getMethod, method, keys, param) {
-        var out, k, len;
-        if (!keys) keys = that.constructor[collection];
-
-        out = {};
-        if (J.isArray(keys)) {
-            k = -1, len = keys.length;
-            for ( ; ++k < len; ) {
-                out[keys[k]] = that[getMethod](keys[k], param);
-            }
-        }
-        else {
-            for (k in keys) {
-                if (keys.hasOwnProperty(k)) {
-                    out[k] = that[getMethod](k, param);
-                }
-            }
-        }
-        return out;
-    }
-
-    /**
-     * ### strSetterMulti
-     *
-     * Same as strSetter, but sets multiple values at once
-     *
-     * @param {object} that The main instance
-     * @param {object} obj List of properties to set and their values
-     * @param {string} collection The name of the collection inside the instance
-     * @param {string} setMethod The name of the method to set each value
-     * @param {string} method The name of the invoking method (for error string)
-     *
-     * @see strSetter
-     */
-    function strSetterMulti(that, obj, collection, setMethod, method) {
-        var i, out;
-        out = out || {};
-        if ('object' !== typeof obj && 'undefined' !== typeof obj) {
-            throw new TypeError(method + ': ' +  collection +
-                                ' must be object or undefined. Found: ' + obj);
-        }
-        for (i in obj) {
-            if (obj.hasOwnProperty(i)) {
-                that[setMethod](i, obj[i]);
-            }
-        }
-    }
-
-    /**
-     * ### strSetter
-     *
-     * Sets the value of a property in a collection if string, function or false
-     *
-     * @param {object} that The main instance
-     * @param {string} name The name of the property to set
-     * @param {string|function|false} value The value for the property
-     * @param {string} collection The name of the collection inside the instance
-     * @param {string} method The name of the invoking method (for error string)
-     *
-     * @see strSetter
-     */
-    function strSetter(that, name, value, collection, method) {
-        if ('undefined' === typeof that.constructor[collection][name]) {
-            throw new TypeError(method + ': name not found: ' + name);
-        }
-        if ('string' === typeof value ||
-            'function' === typeof value ||
-            false === value) {
-
-            that[collection][name] = value;
-        }
-        else {
-            throw new TypeError(method + ': value for item "' + name +
-                                '" must be string, function or false. ' +
-                                'Found: ' + value);
-        }
-    }
-
 })(node);
```

Comment:
You should remove also the property from the constructor: .texts and .sounds are automatically added by the Widgets.js file now.

Reply:
Do you mean the WaitingRoom.text = {...} and WaitingRoom.sounds = {...} properties should be removed?

Reply:
I mean this:

        /**
         * ### WaitingRoom.texts
         *
         * Contains all the texts displayed to the players
         *
         * @see WaitingRoom.setText
         * @see WaitingRoom.getText
         * @see WaitingRoom.setTexts
         * @see WaitingRoom.getTexts
         */
        this.texts = {};

        /**
         * ### WaitingRoom.sounds
         *
         * List of custom sounds to play to the players
         *
         * @see WaitingRoom.setSound
         * @see WaitingRoom.getSound
         * @see WaitingRoom.setSounds
         * @see WaitingRoom.getSounds
         */
         this.sounds = {};

Make sure the @see comments are properly copied in Widget.js